packages/pluggableWidgets/bar-chart-web/src/BarChart.editorConfig.ts

```diff
@@ -0,0 +1,95 @@
+import { BarChartPreviewProps } from "../typings/BarChartProps";
+import {
+    hideNestedPropertiesIn,
+    hidePropertiesIn,
+    hidePropertyIn,
+    Problem,
+    Properties,
+    StructurePreviewProps,
+    transformGroupsIntoTabs
+} from "@mendix/piw-utils-internal";
+
+export function getProperties(
+    values: BarChartPreviewProps,
+    defaultProperties: Properties,
+    platform: "web" | "desktop"
+): Properties {
+    const showAdvancedOptions = values.developerMode !== "basic";
+    values.series.forEach((dataSeries, index) => {
+        if (dataSeries.dataSet === "static") {
+            hideNestedPropertiesIn(defaultProperties, values, "series", index, [
+                "dynamicDataSource",
+                "dynamicXAttribute",
+                "dynamicYAttribute",
+                "dynamicName",
+                "groupByAttribute"
+            ]);
+        } else {
+            hideNestedPropertiesIn(defaultProperties, values, "series", index, [
+                "staticDataSource",
+                "staticXAttribute",
+                "staticYAttribute",
+                "staticName"
+            ]);
+        }
+
+        if (!showAdvancedOptions) {
+            hidePropertyIn(defaultProperties, values, "series", index, "customSeriesOptions");
+        }
+    });
+
+    if (platform === "web") {
+        hidePropertyIn(defaultProperties, values, "developerMode");
+
+        transformGroupsIntoTabs(defaultProperties);
+    } else {
+        if (!showAdvancedOptions) {
+            hidePropertiesIn(defaultProperties, values, ["customLayout", "customConfigurations"]);
+        }
+    }
+
+    return defaultProperties;
+}
+
+export function getPreview(_values: BarChartPreviewProps): StructurePreviewProps | null {
```

Comment:
Just a question, do wee need this method if there is nothing to preview?

Reply:
This is just a copy, I don't aware what the purpose

Reply:
I checked it, so you can even remove this functionality and structure preview will inherit what it has on Studio pro side